modules/gapi/samples/face_beautification.cpp

```diff
@@ -0,0 +1,796 @@
+        //  some operations applied to an input node, so it handles the case
+        //  when it should be nothing to draw
+            gimgShow = cv::gapi::copy(gimgTemp);
+        }
+        return cv::GComputation(cv::GIn(gimgIn),
+                                cv::GOut(gimgBeautif, gimgShow));
+    });
+    // Declaring IE params for networks
+    auto faceParams  = cv::gapi::ie::Params<custom::FaceDetector>
+    {
+        faceXmlPath,
+        faceBinPath,
+        faceDevice
+    };
+    auto landmParams = cv::gapi::ie::Params<custom::LandmDetector>
+    {
+        landmXmlPath,
+        landmBinPath,
+        landmDevice
+    };
+    auto networks      = cv::gapi::networks(faceParams, landmParams);
+    // Declaring custom and fluid kernels have been used:
+    auto customKernels = cv::gapi::kernels<custom::GCPUBilateralFilter,
+                                           custom::GCPULaplacian,
+                                           custom::GCPUFillPolyGContours,
+                                           custom::GCPUPolyLines,
+                                           custom::GCPURectangle,
+                                           custom::GCPUFacePostProc,
+                                           custom::GCPULandmPostProc,
+                                           custom::GCPUGetContours>();
+    auto kernels       = cv::gapi::combine(cv::gapi::core::fluid::kernels(),
+                                           customKernels);
+    // Now we are ready to compile the pipeline to a stream with specified
+    //  kernels, networks and image format expected to process
+    auto stream = pipeline.compileStreaming(cv::GMatDesc{CV_8U,3,
+                                                         cv::Size(640,480)},
+                                            cv::compile_args(kernels,
+                                                             networks));
+    // Setting the source for the stream:
+    if (parser.has("input"))
+    {
+        stream.setSource(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>
+                         (parser.get<cv::String>("input")));
+    }
+    else
+    {
+        stream.setSource(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>
+                         (0));
+    }
+    // Declaring output variables
+    cv::Mat imgShow;
+    cv::Mat imgBeautif;
+    // Streaming:
+    stream.start();
+    while (stream.running())
+    {
+        auto out_vector = cv::gout(imgBeautif, imgShow);
+        if (!stream.try_pull(std::move(out_vector)))
+        {
+            // Use a try_pull() to obtain data.
+            // If there's no data, let UI refresh (and handle keypress)
+            if (cv::waitKey(1) >= 0) break;
+            else continue;
+        }
+        cv::imshow(config::kWinInput,              imgShow);
+        cv::imshow(config::kWinFaceBeautification, imgBeautif);
+    }
+    return 0;
+
+#else
```

Comment:
if this `#else` stands for this if `#if defined(HAVE_OPENCV_GAPI) && defined(HAVE_INF_ENGINE)`, then
1. Please put the appropriate comment - it is hard to track otherwise
2. Breaking the code to have `}` hanging at the last line is wrong. You better declare two distinct `main` functions and cut those with preprocessor fully rather than making reader guess why this `main` is not finished here but new `main` is introduced a couple of lines later.

Reply:
fixed